holmes/plugins/toolsets/opensearch.py

```diff
@@ -0,0 +1,127 @@
+        shards = client.client.cat.shards()
+        return str(shards)
+
+    def get_parameterized_one_liner(self, params:Dict) -> str:
+        return f"opensearch ListShards({params.get('host')})"
+
+class GetClusterSettings(Tool):
+    def __init__(self, opensearch_clients:List[OpenSearchClient]):
+        super().__init__(
+            name = "opensearch_get_cluster_settings",
+            description = "Retrieve the cluster's settings",
+            parameters = {
+                "host": ToolParameter(
+                    description="The cluster host",
+                    type="string",
+                    required=False,
+                )
+            },
+        )
+        self._opensearch_clients = opensearch_clients
+
+    def invoke(self, params:Any) -> str:
+        client = get_client(self._opensearch_clients, host=params.get("host"))
+        response = client.client.cluster.get_settings(
+            include_defaults=True,
+            flat_settings=True
+        )
+        return str(response)
+
+    def get_parameterized_one_liner(self, params) -> str:
+        return f"opensearch GetClusterSettings({params.get('host')})"
+
+
+class GetClusterHealth(Tool):
+    def __init__(self, opensearch_clients:List[OpenSearchClient]):
+        super().__init__(
+            name = "opensearch_get_cluster_health",
+            description = "Retrieve the cluster's health",
+            parameters = {
+                "host": ToolParameter(
+                    description="The cluster host",
+                    type="string",
+                    required=False,
+                )
+            },
+        )
+        self._opensearch_clients = opensearch_clients
+
+    def invoke(self, params:Any) -> str:
+        client = get_client(self._opensearch_clients, host=params.get("host", ""))
+        health = client.client.cluster.health()
+        return str(health)
+
+    def get_parameterized_one_liner(self, params) -> str:
+        return f"opensearch GetClusterSettings({params.get('host')})"
+
+class OpenSearchToolset(Toolset):
+    def __init__(self, clusters_configs:List[Dict]):
+        clients: List[OpenSearchClient] = []
+        for config in clusters_configs:
+            logging.info(f"Setting up OpenSearch client: {str(config)}")
+            client = OpenSearchClient(**config)
+            print(client.client.cluster.health())
+            clients.append(client)
+
+        super().__init__(
+            name = "opensearch",
+            description="Provide cluster metadata information like health, shards, settings.",
+            docs_url="https://opensearch.org/docs/latest/clients/python-low-level/",
+            icon_url="https://upload.wikimedia.org/wikipedia/commons/9/91/Opensearch_Logo.svg",
+            prerequisites = [
+                StaticPrerequisite(enabled=len(clients) > 0, disabled_reason="No opensearch client was configured")
```

Comment:
We don't want to check the client status?

Reply:
made it safer, only add healthy clients to list